dxr/plugins/clang/menus.py

```diff
@@ -201,6 +203,33 @@ def _more_menu_items(self, (qualname,)):
 class FunctionRef(_RefWithDefinition):
     """Ref for function definitions or references"""
 
+    def menu_items(self):
+        """Return a jump-to-definition menu item along with whatever others
+        _more_menu_items() returns.
+
+        """
+        path = self.menu_data[0]
+        if path is None:
+            menu = []
+        else:
+            # We get here when the defloc lies inside the source tree, but for
+            # function refs the defloc can be wrong: defloc may point to the
+            # function declaration instead of the definition if the definition
+            # lies outside the translation unit of the ref.  That means the
+            # query below will return 0 results if the defloc is inside and the
+            # actual def is outside the source tree, and conversely this menu
+            # won't be displayed (even though it should be) if the defloc is
+            # outside and the actual def is inside the source tree.
+            menu = [{'html': "Jump to definition",
+                     'title': "Jump to definition",
+                     'href': url_for('.search', tree=self.tree.name,
+                                     q='+function:"' + self.menu_data[2] + '"',
+                                     redirect='true',
+                                     no_from='true'),
```

Comment:
We're starting to accumulate quite a few special-case querystring params for `search`. Do you see a way we could have fewer? If not, we should document them all in some central place.